### PR TITLE
move s3 copy-from related out to plugin module

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB

refactoring as part of  #11654

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by jeeminso** [comment]
manually tested that a simple copy from s3 scenario works find with the changes

File: app/src/main/java/io/crate/node/CrateNode.java [hunk 1]
```diff
@@ -46,6 +47,7 @@
         UDCPlugin.class,
         URLRepositoryPlugin.class,
         S3RepositoryPlugin.class,
+        S3CopyPlugin.class,
```
Comment (mfussenegger) [inline app/src/main/java/io/crate/node/CrateNode.java:50]: Would it be possible to remove this concrete dependency and rely on the dynamic loading? See the todo:

https://github.com/crate/crate/blob/5e6ffad8bbf32bff9dd4d5e5ff85aa73803d5da5/app/build.gradle#L58-L59

The mechanism here is preferred:

https://github.com/crate/crate/blob/5e6ffad8bbf32bff9dd4d5e5ff85aa73803d5da5/app/build.gradle#L34-L43

File: server/src/main/java/org/elasticsearch/node/Node.java [hunk 2]
```diff
@@ -524,6 +526,9 @@ public Node(final Environment environment,
             );
             modules.add(repositoriesModule);
 
+            FileCollectModule fileCollectModule = new FileCollectModule(pluginsService.filterPlugins(CopyPlugin.class));
```
Comment (mfussenegger) [inline server/src/main/java/org/elasticsearch/node/Node.java:529]: Maybe we rename this to `CopyModule` or so?

**Review by mfussenegger — COMMENTED** [review]
Left some minor comments. Nice work :+1:

**Comment by mfussenegger** [comment]
We should probably also follow up and do the same for COPY TO. S3 is currently hardcoded here: https://github.com/crate/crate/blob/5e6ffad8bbf32bff9dd4d5e5ff85aa73803d5da5/server/src/main/java/io/crate/execution/engine/export/FileWriterCountCollector.java#L96-L100 

We probably want to make a map from scheme to https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/execution/engine/export/Output.java  part of the `CopyPlugin` interface.

File: app/src/main/java/io/crate/node/CrateNode.java (hunk 1 above)
Comment (jeeminso) [reply app/src/main/java/io/crate/node/CrateNode.java:50]: referred to #11307, as the PR explains, it does not work if ran as `app:run` but if I build and run locally, copy from s3 works fine.

**Comment by jeeminso** [comment]
> We should probably also follow up with another PR and do the same for COPY TO. S3 is currently hardcoded here:
> 
> https://github.com/crate/crate/blob/5e6ffad8bbf32bff9dd4d5e5ff85aa73803d5da5/server/src/main/java/io/crate/execution/engine/export/FileWriterCountCollector.java#L96-L100
> 
> We probably want to make a map from scheme to https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/execution/engine/export/Output.java (or some factory for it) part of the `CopyPlugin` interface.

Thanks, I will move onto it right after.

File: app/src/main/java/io/crate/node/CrateNode.java (hunk 1 above)
Comment (mfussenegger) [reply app/src/main/java/io/crate/node/CrateNode.java]: > it does not work if ran as app:run but if I build and run locally, copy from s3 works fine.

yep that's expected. `app:run` doesn't load the plugins.

**Review by mfussenegger — APPROVED** [review]
:+1:

**Pull request merged**